nevergrad/optimization/optimizerlib.py

```diff
@@ -1907,59 +1925,61 @@ def __init__(self, parametrization: IntOrParameter, budget: tp.Optional[int] = N
         self.has_noise = not (descr.deterministic and descr.deterministic_function)
         self.fully_continuous = descr.continuous
         all_params = paramhelpers.flatten_parameter(self.parametrization)
-        self.has_discrete_not_softmax = any(isinstance(x, p.BaseChoice) for x in all_params.values())
-        arity: int = max(len(param.choices) if isinstance(param, p.BaseChoice) else -1 for param in all_params.values())
+        choicetags = [_as_choice_tag(x) for x in all_params.values()]
+        self.has_discrete_not_softmax = any(issubclass(ct.cls, p.BaseChoice) for ct in choicetags)
+        arity: int = max(ct.arity for ct in choicetags)
+        optimClass: base.OptCls
         if self.has_noise and (self.has_discrete_not_softmax or not self.parametrization.descriptors.metrizable):
-            optimClass = RecombiningPortfolioOptimisticNoisyDiscreteOnePlusOne  # type: ignore
+            optimClass = RecombiningPortfolioOptimisticNoisyDiscreteOnePlusOne
```

Comment:
When a `# type: ignore` needs to be added at every line with optimClass, it means `optimClass` was not typed properly, please ping me instead of adding it everywhere.